File: workspace/contest/codeforce394D.py
if __name__ == '__main__':
	n=int(input())
	height=[int(i) for i in input().split()]
	n=len(height)
	height.sort()
	maxStep=(height[-1]-height[0])
	minCost=999999
	stepAns=0
	start=0
	#print(height)
	#print(maxStep)
	for step in range(0,maxStep+1):
		maxDiff=-999999
		minDiff= 999999
		for i in range(0,n):
			maxDiff=max(maxDiff,height[i]-step*i)
			minDiff=min(minDiff,height[i]-step*i)
		mid=(maxDiff+minDiff)//2
		#print("max: {0} min: {1} with step:{2}".format(maxDiff,minDiff,step))
		cost=max(maxShift-mid, mid-minShift)
		if cost<minCost:
			minCost=cost
			start=mid
			stepAns=step
			#print("update ans for cost:{0} start:{1} step:{2}".format(minCost,start,stepAns))
	print(minCost)
	print("{0} {1}".format(start,stepAns))
	
'''
6
94 65 -33 -43 60 -24

'''

File: workspace/contest/codeforce394D.cpp
#include <iostream>
#include <cstdlib>
using namespace std;
#ifndef MAX_N
#define MAX_N 1010
#endif
int n;
int height[MAX_N];
int minCost=999999;
int maxStep,stepAns,start;
int cmp (const void * a, const void * b)
{
  if ( *(int*)a <  *(int*)b ) return -1;
  if ( *(int*)a == *(int*)b ) return 0;
  if ( *(int*)a >  *(int*)b ) return 1;
}
int main(int argc, char const *argv[])
{
	cin>>n;
	for (int i = 0; i < n; ++i){
		cin>>height[i];
	}
	qsort(height,n,sizeof(int),cmp);
	maxStep=height[n-1]-height[0];
	for (int step = 0; step <=maxStep; ++step){
		int maxShift=-999999,minShift= 999999;
		for (int i = 0; i < n; ++i){
			maxShift=max(maxShift,height[i]-step*i);
			minShift=min(minShift,height[i]-step*i);
		}
		int mid=(maxShift+minShift)/2;
		int cost=max(maxShift-mid, mid-minShift);
		if ( cost < minCost ){
			minCost=cost;
			start=mid;
			stepAns=step;
		}
	}
	cout<<minCost<<endl;
	cout<<start<<" "<<stepAns;
	return 0;
}
/*
6
94 65 -33 -43 60 -24
*/